Custom Mappings and Custom Sortable Fields List (issue #579) (#643)
stac-utils/stac-fastapi-elasticsearch: README.md

### Environment Variables

| Variable | Index | Description | Default |
|----------|------|-------------|---------|
| `STAC_FASTAPI_ES_CUSTOM_MAPPINGS` | items | JSON string of property mappings to merge with defaults | None |
| `STAC_FASTAPI_ES_MAPPINGS_FILE` | items | Path to a JSON file containing property mappings to merge with defaults | None |

> | `STAC_FASTAPI_ES_COLLECTIONS_MAPPINGS` | collections| JSON string of property mappings to merge with defaults | None |

Reviewer: I think this should be:`STAC_FASTAPI_ES_COLLECTIONS_CUSTOM_MAPPINGS`

| `STAC_FASTAPI_ES_COLLECTIONS_MAPPINGS_FILE` | collections| Path to a JSON file containing property mappings to merge with defaults | None |
| `STAC_FASTAPI_ES_CUSTOM_DYNAMIC_TEMPLATES` | dynamic template| JSON string of templates to merge with defaults | None |
| `STAC_FASTAPI_ES_DYNAMIC_TEMPLATES_FILE` | dynamic template| Path to a JSON file containing templates to merge with defaults | None |
| `STAC_FASTAPI_ES_DYNAMIC_MAPPING` | dynamic mapping | Controls dynamic mapping: `true`, `false`, or `strict` | `true` |

### Custom Mappings

You can customize the Elasticsearch/OpenSearch mappings by providing a JSON configuration. This can be done via:

1. `STAC_FASTAPI_ES_CUSTOM_MAPPINGS` | `STAC_FASTAPI_ES_COLLECTIONS_MAPPINGS`|`STAC_FASTAPI_ES_DYNAMIC_TEMPLATES` environment variable (takes precedence)

> 2. `STAC_FASTAPI_ES_MAPPINGS_FILE`| `STAC_FASTAPI_ES_COLLECTIONS_MAPPINGS_FILE`| `STAC_FASTAPI_ES_DYNAMIC_TEMPLATES_FILE` environment variable (file path)

Reviewer: ```suggestion
1. `STAC_FASTAPI_ES_CUSTOM_MAPPINGS`, `STAC_FASTAPI_ES_COLLECTIONS_CUSTOM_MAPPINGS`, or `STAC_FASTAPI_ES_CUSTOM_DYNAMIC_TEMPLATES` (takes precedence)
2. `STAC_FASTAPI_ES_MAPPINGS_FILE`, `STAC_FASTAPI_ES_COLLECTIONS_MAPPINGS_FILE`, or `STAC_FASTAPI_ES_DYNAMIC_TEMPLATES_FILE` (file path)
```

# Reference the file
export STAC_FASTAPI_ES_MAPPINGS_FILE=/path/to/custom-mappings.json
```

> Similar approach can be taken for `STAC_FASTAPI_ES_COLLECTIONS_MAPPINGS_FILE` and  `STAC_FASTAPI_ES_DYNAMIC_TEMPLATES_FILE` with json file names `custom-collections-mappings.json` and `custom-dynamic-templates.json` as an example

Reviewer: ```suggestion
A similar approach can be taken for `STAC_FASTAPI_ES_COLLECTIONS_MAPPINGS_FILE` and `STAC_FASTAPI_ES_DYNAMIC_TEMPLATES_FILE`.
```
